File: Documentation/5.4/Samples/python/ClientApi/Operations/HowTo/SwitchOperationsToDifferentDatabase.py
from ravendb import MaintenanceOperationExecutor, DocumentStore, GetStatisticsOperation
from ravendb.documents.operations.executor import OperationExecutor
from ravendb.documents.operations.revisions import GetRevisionsOperation

from examples_base import ExampleBase, Order, Company


class SwitchOperationsToDifferentDatabase(ExampleBase):
    def setUp(self):
        super().setUp()

    class Foo:
        # region syntax_1
        def for_database(self, database_name: str) -> OperationExecutor: ...

        # endregion
        # region syntax_2
        def for_database(self, database_name: str) -> MaintenanceOperationExecutor: ...

        # endregion

    def test_switch_operations_to_different_database(self):
        # region for_database_1
        # Define default database on the store
        document_store = DocumentStore(urls=["yourServerURL"], database="DefaultDB")
        document_store.initialize()

        with document_store:
            # Use 'for_database', get operation executor for another database
            op_executor = document_store.operations.for_database("AnotherDB")

            # Send the operation, e.g. 'GetRevisionsOperation' will be executed on "AnotherDB"
            revisions_in_another_db = op_executor.send(GetRevisionsOperation("Orders/1-A", Order))

            # Without 'for_database', the operation is executed "DefaultDB"
            revisions_in_default_db = document_store.operations.send(GetRevisionsOperation("Company/1-A", Company))
        # endregion

        # region for_database_2
        # Define default database on the store
        document_store = DocumentStore(urls=["yourServerURL"], database="DefaultDB")
        document_store.initialize()

        with DocumentStore() as document_store:
            # Use 'for_database', get maintenance operation executor for another database
            op_executor = document_store.maintenance.for_database("AnotherDB")
            # Send the maintenance operation, e.g. get database stats for "AnotherDB"
            stats_for_another_db = op_executor.send(GetStatisticsOperation())
            # Without 'for_database', the stats are retrieved for "DefaultDB"
            stats_for_default_db = document_store.maintenance.send(GetStatisticsOperation())
        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/HowTo/SwitchOperationsToDifferentDatabase.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\HowTo;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\GetStatisticsOperation;
use RavenDB\Documents\Operations\MaintenanceOperationExecutor;
use RavenDB\Documents\Operations\OperationExecutor;
use RavenDB\Documents\Operations\Revisions\GetRevisionsOperation;
use RavenDB\Samples\Infrastructure\Orders\Company;
use RavenDB\Samples\Infrastructure\Orders\Order;

class SwitchOperationsToDifferentDatabase
{
    public function SwitchOperationToDifferentDatabase() : void
    {
        # region for_database_1
        // Define default database on the store
        $documentStore = new DocumentStore(
            ["yourServerURL"],
            "DefaultDB"
        );
        $documentStore->initialize();

        try {
            // Use 'ForDatabase', get operation executor for another database
            /** @var OperationExecutor $opExecutor */
            $opExecutor = $documentStore->operations()->forDatabase("AnotherDB");

            // Send the operation, e.g. 'GetRevisionsOperation' will be executed on "AnotherDB"
            $revisionsInAnotherDB = $opExecutor->send(new GetRevisionsOperation(Order::class, "Orders/1-A"));

            // Without 'ForDatabase', the operation is executed "DefaultDB"
            $revisionsInDefaultDB = $documentStore->operations()->send(new GetRevisionsOperation(Company::class, "Company/1-A"));
        } finally {
            $documentStore->close();
        }
        # endregion
    }

    public function SwitchMaintenanceOperationToDifferentDatabase(): void
    {
        # region for_database_2
        // Define default database on the store
        $documentStore = new DocumentStore(
            [ "yourServerURL" ],
            "DefaultDB"
        );
        $documentStore->initialize();

        try {
            // Use 'ForDatabase', get maintenance operation executor for another database
            /** @var MaintenanceOperationExecutor $opExecutor */
            $opExecutor = $documentStore->maintenance()->forDatabase("AnotherDB");

            // Send the maintenance operation, e.g. get database stats for "AnotherDB"
            $statsForAnotherDB = $opExecutor->send(new GetStatisticsOperation());

            // Without 'ForDatabase', the stats are retrieved for "DefaultDB"
            $statsForDefaultDB = $documentStore->maintenance()->send(new GetStatisticsOperation());
        } finally {
            $documentStore->close();
        }
        # endregion
    }
}

/*
interface OperationsForDatabaseSyntax
{
    # region syntax_1
    public function forDatabase(?string $databaseName): OperationExecutor;
    # endregion
}
*/

/*
interface MaintenanceForDatabaseSyntax
{
    # region syntax_2
    public function forDatabase(?string $databaseName): MaintenanceOperationExecutor;
    # endregion
}
*/
